File: pandoc-anglequotes.hs
import Text.Pandoc.JSON

replaceQuotes ('<':'<':tl) = '«' : replaceQuotes tl
replaceQuotes ('>':'>':tl) = '»' : replaceQuotes tl
replaceQuotes (hd:tl)      = hd  : replaceQuotes tl
replaceQuotes ""           = ""

process (Str text) = Str $ replaceQuotes text
process x = x

main = toJSONFilter process

File: pandoc-anglequotes.py
#!/usr/bin/env python
# coding: utf-8

from __future__ import unicode_literals

from pandocfilters import toJSONFilter, Str


def process(key, value, format, meta, Str=Str):
    if key == "Str":
        return Str(value.replace("<<", '«').replace(">>", '»'))


if __name__ == "__main__":
    toJSONFilter(process)
